editor/plugins/script_editor_plugin.cpp

```diff
@@ -1292,18 +1297,21 @@ void ScriptEditor::_menu_option(int p_option) {
 				save_current_script();
 			} break;
 			case FILE_SAVE_AS: {
-				if (trim_trailing_whitespace_on_save) {
-					current->trim_trailing_whitespace();
-				}
-
-				current->insert_final_newline();
+				if (!format_on_save) {
+					if (trim_trailing_whitespace_on_save) {
+						current->trim_trailing_whitespace();
+					}
+					current->insert_final_newline();
 
-				if (convert_indent_on_save) {
-					if (use_space_indentation) {
-						current->convert_indent_to_spaces();
-					} else {
-						current->convert_indent_to_tabs();
+					if (convert_indent_on_save) {
+						if (use_space_indentation) {
+							current->convert_indent_to_spaces();
+						} else {
+							current->convert_indent_to_tabs();
+						}
 					}
+				} else {
+					current->format_code();
```

Comment:
I know that you only wrapped existing code with an `if` condition, but as this code is repeating itself multiple time in the file, maybe you could create a function that "formats" the code.